leafletwidget/leafletwidget.js

```diff
@@ -351,45 +346,55 @@ require(["widgets/js/widget", "leaflet", "leaflet_draw"], function(WidgetManager
         },
 
     });
-    WidgetManager.register_widget_view('LeafletDrawControlView', LeafletDrawControlView);
+    manager.WidgetManager.register_widget_view('LeafletDrawControlView', LeafletDrawControlView);
 
 
-    var LeafletMapView = IPython.DOMWidgetView.extend({
+    var LeafletMapView = widget.DOMWidgetView.extend({
         
-        initialize: function (options) {
-            LeafletMapView.__super__.initialize.apply(this, arguments);
-            this.rendered = false;
+        initialize: function() {
+            console.log('MapView initialized');
+            LeafletMapView.__super__.initialize.apply(this,arguments);
+            this.rendered=false;
         },
     
         // Layer management
         update_layers: function (old_list, new_list) {
```

Comment:
Ok, here's where I'm struggling. I saw a conversation about deprecating `do_diff` and adding `promises`, but I'm not sure how to update this critical bit of code.